pkg/build/builder/cmd/builder.go

```diff
@@ -67,6 +73,45 @@ func run(builderFactory factoryFunc) {
 
 }
 
+func setupSourceSecret() (err error) {
+	// TODO: remove this when permissions for secrets will be fixed upstream
+	secretContents, err := ioutil.ReadFile(os.Getenv("SOURCE_SECRET_PATH"))
+	if err != nil {
+		return
+	}
+	secretFile, err := ioutil.TempFile("", "sourcesecret")
+	if err != nil {
+		return
+	}
+	defer secretFile.Close()
+	if err = secretFile.Chmod(0600); err != nil {
+		return
+	}
+	if _, err = secretFile.Write(secretContents); err != nil {
+		return
+	}
+	os.Setenv("SOURCE_SECRET_PATH", secretFile.Name())
```

Comment:
Currently secrets are mounted with too broad permissions (755 or alike) the above TODO is addressing that problem, once we fix that upstream, we'll remove that code.